store/sql_post_store.go

```diff
@@ -471,6 +473,8 @@ func (s SqlPostStore) Search(teamId string, userId string, terms string, isHasht
 				    ORDER BY CreateAt DESC
 				LIMIT 100`, searchType)
 
+			terms = strings.TrimRight(terms, specialSearchCharCutset)
```

Comment:
You're correct but MySQL ignores all non-word characters in it's searches so "(hello" is the same as "hello" anyways. MySQL just errors out if one of those speical characters is at the end of the string because it's expecting something more.

"(hello)corey" would stay the same as TrimRight only trims characters from the suffix.